tools/ob_admin/backup_tool/ob_admin_backup_validation_task.cpp

```diff
@@ -0,0 +1,4275 @@
+  } else if (OB_FAIL(ADD_DAG_WARN_INFO_PARAM(out_param, allocator, get_type(),
+                                             static_cast<int64_t>(OB_SERVER_TENANT_ID)))) {
+    STORAGE_LOG(WARN, "failed to fill info param", K(ret));
+  }
+  return ret;
+}
+int ObAdminFinishLogArchiveValidationDag::fill_dag_key(char *buf, const int64_t buf_len) const
+{
+  int ret = OB_SUCCESS;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+  }
+  return ret;
+}
+int64_t ObAdminFinishLogArchiveValidationDag::hash() const
+{
+  int64_t ptr = reinterpret_cast<int64_t>(this);
+  return common::murmurhash(&ptr, sizeof(ptr), 0);
+}
+int ObAdminFinishLogArchiveValidationDag::init(int64_t id, ObAdminBackupValidationCtx *ctx)
+{
+  int ret = OB_SUCCESS;
+  if (IS_INIT) {
+    ret = OB_INIT_TWICE;
+    STORAGE_LOG(WARN, "init twice", K(ret));
+  } else if (OB_ISNULL(ctx)) {
+    ret = OB_INVALID_ARGUMENT;
+    STORAGE_LOG(WARN, "ctx is null", K(ret));
+  } else {
+    is_inited_ = true;
+    id_ = id;
+    ctx_ = ctx;
+  }
+  return ret;
+}
+int ObAdminFinishLogArchiveValidationDag::create_first_task()
+{
+  int ret = OB_SUCCESS;
+  ObAdminFinishLogArchiveValidationTask *task = nullptr;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret));
+  } else if (OB_FAIL(alloc_task(task))) {
+    STORAGE_LOG(WARN, "failed to alloc task", K(ret));
+  } else if (OB_FAIL(task->init(0, ctx_))) {
+    STORAGE_LOG(WARN, "failed to init task", K(ret));
+  } /*add task child here*/ else if (OB_FAIL(add_task(*task))) {
+    STORAGE_LOG(WARN, "failed to add task", K(ret));
+  } else {
+    STORAGE_LOG(INFO, "succeed to create first task", K(ret));
+  }
+  return ret;
+}
+////////////////ObAdminFinishLogArchiveValidationDag End////////////////
+
+////////////////ObAdminFinishLogArchiveValidationTask Start////////////////
+ObAdminFinishLogArchiveValidationTask::~ObAdminFinishLogArchiveValidationTask() {}
+int ObAdminFinishLogArchiveValidationTask::init(int64_t task_id, ObAdminBackupValidationCtx *ctx)
+{
+  int ret = OB_SUCCESS;
+  if (IS_INIT) {
+    ret = OB_INIT_TWICE;
+    STORAGE_LOG(WARN, "init twice", K(ret));
+  } else if (OB_ISNULL(ctx)) {
+    ret = OB_INVALID_ARGUMENT;
+    STORAGE_LOG(WARN, "ctx is null", K(ret));
+  } else {
+    is_inited_ = true;
+    task_id_ = task_id;
+    ctx_ = ctx;
+  }
+  return ret;
+}
+int ObAdminFinishLogArchiveValidationTask::process()
+{
+  int ret = OB_SUCCESS;
+  if (IS_NOT_INIT) {
+    ret = OB_NOT_INIT;
+    STORAGE_LOG(WARN, "not init", K(ret));
+  } else if (OB_ISNULL(ctx_) || ctx_->aborted_) {
+    ret = OB_CANCELED;
+    STORAGE_LOG(WARN, "aborted", K(ret));
+  } else if (OB_FAIL(cross_check_scn_continuity_())) {
+    STORAGE_LOG(WARN, "cross check scn continuity", K(ret));
+  } else {
+    STORAGE_LOG(INFO, "succeed to process ObAdminFinishLogArchiveValidationTask", K(ret),
+                K(task_id_));
+  }
+  return ret;
+}
+int ObAdminFinishLogArchiveValidationTask::cross_check_scn_continuity_()
```

Comment:
the function is too large here, consider split into smaller pieces

Reply:
fixed. split out inner_get_scn_range_ functions